Firestore/core/src/firebase/firestore/remote/serializer.cc

```diff
@@ -612,7 +615,13 @@ google_firestore_v1_Write Serializer::EncodeMutation(
       result.which_operation = google_firestore_v1_Write_update_tag;
       auto patch_mutation = static_cast<const PatchMutation&>(mutation);
       result.update = EncodeDocument(mutation.key(), patch_mutation.value());
-      result.update_mask = EncodeFieldMask(patch_mutation.mask());
+      // Note: the fact that this field is set (even if the mask is empty) is
```

Comment:
At least, that's my understanding based on our conversation -- let me know if I got the details wrong.